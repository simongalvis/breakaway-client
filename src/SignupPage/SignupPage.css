
form{
    
    padding: 16px 0;
    margin-right:0;
    text-align:0;
    display: block;
    background-color:rgb(41, 41, 41);
    max-width:500px;
    
}
input{
    width:80%;
    margin:3%;
    border-radius: 7px;
    border:solid 1px gray;
    height:21px;
    background-color:black;
    max-width:700px;
}
#signupheading{
    color:white;
    background-color: black;
    margin:0;
    padding: 24px 0;
}
#signupbutton{
    width: 80%;
    margin-top:4%;
    border-radius:33px;;
    background-color:rgb(13, 146, 255);
    color:white;
    border: 1px solid rgb(41, 41, 41);
}
