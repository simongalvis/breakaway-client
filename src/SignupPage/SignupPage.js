import React from 'react';
import '../SignupPage/SignupPage.css';
import NavBar from '../NavBar/NavBar.js';
import ApiContext from '../ApiContext';



 


class SignupPage extends React.Component{
static contextType = ApiContext;

    handleSubmit = e =>{
        e.preventDefault();

       const  { signupFullname, signupUsername, signupPassword } = e.target;

       //console.log(` signupFullname: ` +signupFullname.value +  `signupUsername: ` + signupUsername.value + `signupPassword: ` + signupPassword.value)


       const user = {
           fullname: signupFullname.value,
           username: signupUsername.value,
           password: signupPassword.value
       }
       this.context.createUser(user)
       this.context.refreshDatabase()
       alert('Account successfully created. Now Click the login button on the top right of the page to login and get started!')

       
    }

     
    
    
    render(){
        return(
            <div className='SignupPage'>
                <NavBar/>
                <div className='signupFormContainer'>
                <h2 id='signupheading'>Become part of the Breakaway crew</h2>
                <form onSubmit={this.handleSubmit}> 
                    <input  type='text' id='signupFullname' name='signupFullname' placeholder='Full name' required/>
                    <input  type='text' id='signupUsername' name='signupUsername' placeholder='Username' required/>
                    <input  type='text' id='signupPassword' name='signupPassword' placeholder='Password' required/>
                    <button id='signupbutton'>Sign Up</button>
                    
                    
                </form>
                </div>
                
            </div>
            
        )
    }
}

export default SignupPage;