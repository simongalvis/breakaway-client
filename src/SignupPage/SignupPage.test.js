import React from 'react'
import ReactDOM from 'react-dom'
import SignupPage from './SignupPage'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`SignupPage component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<BrowserRouter><SignupPage/></BrowserRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<BrowserRouter><SignupPage/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
