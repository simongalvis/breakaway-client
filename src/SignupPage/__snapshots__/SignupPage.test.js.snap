// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage component Renders UI as expected 1`] = `
<div
  className="SignupPage"
>
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Blue breakaway logo"
        id="main-logo"
        src="breakaway-logo.png"
      />
    </a>
    <a
      href="/login-page"
      onClick={[Function]}
    >
      <button
        className="nav-button"
        id="login-button"
        value="login"
      >
        Log In
      </button>
    </a>
  </header>
  <div
    className="signupFormContainer"
  >
    <h2
      id="signupheading"
    >
      Become part of the Breakaway crew
    </h2>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="signupFullname"
      >
        <b>
          Full name:
        </b>
      </label>
      <input
        autoCapitalize="none"
        autoCorrect="off"
        id="signupFullname"
        name="signupFullname"
        placeholder="Full name"
        required={true}
        type="text"
      />
      <label
        htmlFor="signupUsername"
      >
        <b>
          Username:
        </b>
      </label>
      <input
        autoCapitalize="none"
        autoCorrect="off"
        id="signupUsername"
        name="signupUsername"
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        htmlFor="signupPassword"
      >
        <b>
          Password:
        </b>
      </label>
      <input
        autoCapitalize="none"
        autoCorrect="off"
        id="signupPassword"
        name="signupPassword"
        placeholder="Password"
        required={true}
        type="text"
      />
      <button
        id="signupbutton"
      >
        <b>
          Sign Up
        </b>
      </button>
    </form>
  </div>
</div>
`;
